Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/CapsuleViewportEdit.cpp

```diff
@@ -0,0 +1,205 @@
+            m_heightManipulator->SetViews(AZStd::move(views));
+        }
+        m_heightManipulator->InstallMouseMoveCallback([this]
+        (const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                OnHeightManipulatorMoved(action);
+            });
+    }
+
+    void CapsuleViewportEdit::OnRadiusManipulatorMoved(const AzToolsFramework::LinearManipulator::Action& action)
+    {
+        // manipulator action offsets do not take entity transform scale into account, so need to apply it here
+        const AZ::Transform capsuleLocalTransform = GetCapsuleLocalTransform();
+        const AZ::Vector3 manipulatorPosition = AzToolsFramework::GetPositionInManipulatorFrame(
+            m_radiusManipulator->GetSpace().GetUniformScale(), capsuleLocalTransform, action);
+
+        // Get the distance the manipulator has moved along the axis.
+        float extent = manipulatorPosition.Dot(action.m_fixed.m_axis);
+
+        // Clamp radius to a small value.
+        extent = AZ::GetMax(extent, MinCapsuleRadius);
+
+        // Update the manipulator and capsule radius.
+        m_radiusManipulator->SetLocalTransform(capsuleLocalTransform * AZ::Transform::CreateTranslation(extent * action.m_fixed.m_axis));
+
+        // Adjust the height manipulator so it is always clamped to twice the radius.
+        AdjustHeightManipulator(extent);
+
+        // The final radius of the capsule is the manipulator's extent.
+        SetCapsuleRadius(extent);
+    }
+
+    void CapsuleViewportEdit::OnHeightManipulatorMoved(const AzToolsFramework::LinearManipulator::Action& action)
+    {
+        // manipulator action offsets do not take entity transform scale into account, so need to apply it here
+        const AZ::Transform localTransform = GetCapsuleLocalTransform();
+        const AZ::Vector3 manipulatorPosition = AzToolsFramework::GetPositionInManipulatorFrame(
+            m_heightManipulator->GetSpace().GetUniformScale(), localTransform, action);
+
+        // Get the distance the manipulator has moved along the axis.
+        float extent = manipulatorPosition.Dot(action.m_fixed.m_axis);
+
+        // Ensure capsule's half height is always greater than the radius.
+        extent = AZ::GetMax(extent, MinCapsuleHeight);
+
+        // Update the manipulator and capsule height.
+        m_heightManipulator->SetLocalTransform(localTransform * AZ::Transform::CreateTranslation(extent * action.m_fixed.m_axis));
+
+        // The final height of the capsule is twice the manipulator's extent.
+        float capsuleHeight = extent / HalfHeight;
+
+        // Adjust the radius manipulator so it is always clamped to half the capsule height.
+        AdjustRadiusManipulator(capsuleHeight);
+
+        // Finally adjust the capsule height
+        SetCapsuleHeight(capsuleHeight);
+    }
+
+    void CapsuleViewportEdit::AdjustRadiusManipulator(const float capsuleHeight)
```

Comment:
Nit: no const for integral parameters according to coding std

Reply:
Oh yeah, think that came from the original code I moved, but I'll update it.

Reply:
Fixed.